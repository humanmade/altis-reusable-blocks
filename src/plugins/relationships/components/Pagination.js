import PropTypes from 'prop-types';

import { PanelRow } from '@wordpress/components';
import { __ } from '@wordpress/i18n';

const Pagination = ( {
	currentPage,
	goToPrevPage,
	goToNextPage,
	totalPages,
	totalItems,
} ) => {
	if ( totalPages < 2 ) {
		return null;
	}

	return (
		<PanelRow className="tablenav relationship-pagination">
			<div className="tablenav-pages">
				<span className="displaying-num">{ sprintf( __( '%d items', 'enhanced-reusable-blocks' ), totalItems ) }</span>
				<span className="pagination-links">
					{ currentPage === 1 ?
						<span className="tablenav-pages-navspan button disabled" aria-hidden="true">‹</span>
						: 	<a className="prev-page button" onClick={ goToPrevPage }>
								<span className="screen-reader-text">{ __( 'Previous page', 'enhanced-reusable-blocks' ) }</span>
								<span aria-hidden="true">‹</span>
							</a>
					}
					<span className="screen-reader-text">{ __( 'Current Page', 'enhanced-reusable-blocks' ) }</span>
					<span id="table-paging" className="paging-input">
						<span className="tablenav-paging-text">{ sprintf( __( '%d of %d', 'enhanced-reusable-blocks' ), currentPage, totalPages ) } </span>
					</span>
					{ currentPage === totalPages ?
						<span className="tablenav-pages-navspan button disabled" aria-hidden="true">›</span>
						: 	<a className="next-page button" onClick={ goToNextPage }>
								<span className="screen-reader-text">{ __( 'Next page', 'enhanced-reusable-blocks' ) }</span>
								<span aria-hidden="true">›</span>
							</a>
					}
				</span>
			</div>
		</PanelRow>
	);
};

Pagination.propTypes = {
	currentPage: PropTypes.number.isRequired,
	goToPrevPage: PropTypes.func.isRequired,
	goToNextPage: PropTypes.func.isRequired,
	totalPages: PropTypes.number.isRequired,
	totalItems: PropTypes.number.isRequired,
};

export default Pagination;
