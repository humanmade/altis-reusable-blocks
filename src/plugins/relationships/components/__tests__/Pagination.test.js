import { shallow } from 'enzyme';

import Pagination from '../Pagination';

jest.mock( '@wordpress/components', () => mockComponents( [
	'PanelRow',
] ) );

describe( '<Pagination />', () => {

	test.each( [
		[ 0 ],
		[ 1 ],
	] )( 'renders nothing if total number of pages is %d', ( totalPages ) => {
		const result = shallow(
			<Pagination
				currentPage={ 1 }
				goToNextPage={ () => {} }
				goToPrevPage={ () => {} }
				totalItems={ totalPages }
				totalPages={ totalPages }
			/>
		);

		expect( result ).toMatchSnapshot();
	} );

	test( 'renders as expected if current page is first page', () => {
		const result = shallow(
			<Pagination
				currentPage={ 1 }
				goToNextPage={ () => {} }
				goToPrevPage={ () => {} }
				totalItems={ 42 }
				totalPages={ 5 }
			/>
		);

		expect( result ).toMatchSnapshot();
	} );

	test( 'renders as expected if current page is last page', () => {
		const result = shallow(
			<Pagination
				currentPage={ 5 }
				goToNextPage={ () => {} }
				goToPrevPage={ () => {} }
				totalItems={ 42 }
				totalPages={ 5 }
			/>
		);

		expect( result ).toMatchSnapshot();
	} );

	test( 'renders as expected if current page is neither first nor last page', () => {
		const result = shallow(
			<Pagination
				currentPage={ 3 }
				goToNextPage={ () => {} }
				goToPrevPage={ () => {} }
				totalItems={ 42 }
				totalPages={ 5 }
			/>
		);

		expect( result ).toMatchSnapshot();
	} );

} );
