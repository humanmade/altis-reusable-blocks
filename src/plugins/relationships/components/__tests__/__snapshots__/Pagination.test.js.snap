// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> renders as expected if current page is first page 1`] = `
<PanelRow
  className="tablenav relationship-pagination"
>
  <div
    className="tablenav-pages"
  >
    <span
      className="displaying-num"
    >
      42 items
    </span>
    <span
      className="pagination-links"
    >
      <span
        aria-hidden="true"
        className="tablenav-pages-navspan button disabled"
      >
        ‹
      </span>
      <span
        className="screen-reader-text"
      >
        Current Page
      </span>
      <span
        className="paging-input"
        id="table-paging"
      >
        <span
          className="tablenav-paging-text"
        >
          1 of 5
        </span>
      </span>
      <button
        className="next-page button"
        onClick={[Function]}
      >
        <span
          className="screen-reader-text"
        >
          Next page
        </span>
        <span
          aria-hidden="true"
        >
          ›
        </span>
      </button>
    </span>
  </div>
</PanelRow>
`;

exports[`<Pagination /> renders as expected if current page is last page 1`] = `
<PanelRow
  className="tablenav relationship-pagination"
>
  <div
    className="tablenav-pages"
  >
    <span
      className="displaying-num"
    >
      42 items
    </span>
    <span
      className="pagination-links"
    >
      <button
        className="prev-page button"
        onClick={[Function]}
      >
        <span
          className="screen-reader-text"
        >
          Previous page
        </span>
        <span
          aria-hidden="true"
        >
          ‹
        </span>
      </button>
      <span
        className="screen-reader-text"
      >
        Current Page
      </span>
      <span
        className="paging-input"
        id="table-paging"
      >
        <span
          className="tablenav-paging-text"
        >
          5 of 5
        </span>
      </span>
      <span
        aria-hidden="true"
        className="tablenav-pages-navspan button disabled"
      >
        ›
      </span>
    </span>
  </div>
</PanelRow>
`;

exports[`<Pagination /> renders as expected if current page is neither first nor last page 1`] = `
<PanelRow
  className="tablenav relationship-pagination"
>
  <div
    className="tablenav-pages"
  >
    <span
      className="displaying-num"
    >
      42 items
    </span>
    <span
      className="pagination-links"
    >
      <button
        className="prev-page button"
        onClick={[Function]}
      >
        <span
          className="screen-reader-text"
        >
          Previous page
        </span>
        <span
          aria-hidden="true"
        >
          ‹
        </span>
      </button>
      <span
        className="screen-reader-text"
      >
        Current Page
      </span>
      <span
        className="paging-input"
        id="table-paging"
      >
        <span
          className="tablenav-paging-text"
        >
          3 of 5
        </span>
      </span>
      <button
        className="next-page button"
        onClick={[Function]}
      >
        <span
          className="screen-reader-text"
        >
          Next page
        </span>
        <span
          aria-hidden="true"
        >
          ›
        </span>
      </button>
    </span>
  </div>
</PanelRow>
`;

exports[`<Pagination /> renders nothing if total number of pages is 0 1`] = `""`;

exports[`<Pagination /> renders nothing if total number of pages is 1 1`] = `""`;
